app/controllers/auth_controller.rb

```diff
@@ -30,4 +36,32 @@ def auth_with_okta
     render_bad_request e.message
     nil
   end
+
+  def auth_with_facebook
+    return unless /^[0-9a-zA-Z]/.match?(data_attrs[:facebook_access_token])
+    url = "https://graph.facebook.com/debug_token?input_token=#{data_attrs[:facebook_access_token]}&access_token=#{ENV.fetch("FACEBOOK_APP_ID")}|#{ENV.fetch("FACEBOOK_APP_SECRET")}"
+
+    begin
+      data = JSON.parse(Net::HTTP.get(URI(url)))
+    rescue Net::HTTPBadRequest => e
+      render_bad_request(e.to_s) and return
+    end
+
+    (render_bad_request(data["data"]["error"]) and return) if data["data"] && data["data"]["error"]
+    (render_bad_request("facebook token is not valid") and return) unless data["data"] && data["data"]["is_valid"] && data["data"]["user_id"]
+
+    user_id = data["data"]["user_id"]
+    url = "https://graph.facebook.com/#{user_id}?fields=email,id,first_name,last_name,short_name&access_token=#{data_attrs[:facebook_access_token]}"
```

Comment:
you will need to include the secret here, dev currently has the "require app secret" setting disabled, but we will want that enabled, and it is currently enabled on production.

Reply:
Surprisingly, it works in production without app secret. I just tested it.

Reply:
ok, guess we don't need to do anything different here then